linera-core/src/client.rs

```diff
@@ -1328,6 +1335,59 @@ where
                         message.action = MessageAction::Reject;
                         continue;
                     }
+                } else if let ChainError::ExecutionError(
+                    ExecutionError::SystemError(SystemExecutionError::BlobNotFoundOnRead(blob_id)),
+                    _,
```

Comment:
We should probably also try to fetch the missing blobs in the `ChainExecutionContext::IncomingMessage(index)` case, and reject that message only after this has failed. (Doesn't apply in this PR yet, because messages can't read blobs yet.)

So we could leave `stage_block_execution_and_discard_failing_messages` as it is, and move _this_ logic into an inner `self.stage_block_execution(block)` call (which in turn calls `self.client.local_node.stage_block_execution(block.clone())`). But as discussed, we will ultimately want to fetch the blobs without restarting execution from the beginning.

Reply:
Is this all then for a separate PR after messages are able to read blobs then? 🤔

Reply:
If you prefer, but we have to make sure we remember it. The new system operation is meant to simulate what in production will actually be user operations and incoming messages. So I'm a bit worried about any code that _only_ makes the operation work, but not messages, because then our tests give us a false sense of security.

Reply:
Ok! Makes sense, I can just do it on this PR then